include/range/v3/view/trim.hpp

```diff
@@ -0,0 +1,147 @@
+/// \file
+// Range v3 library
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_TRIM_HPP
+#define RANGES_V3_VIEW_TRIM_HPP
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+#include <concepts/concepts.hpp>
+#include <range/v3/algorithm/find_if_not.hpp>
+#include <range/v3/detail/config.hpp>
+#include <range/v3/iterator/concepts.hpp>
+#include <range/v3/functional/bind.hpp>
+#include <range/v3/functional/compose.hpp>
+#include <range/v3/functional/pipeable.hpp>
+#include <range/v3/range/access.hpp>
+#include <range/v3/range/concepts.hpp>
+#include <range/v3/range/primitives.hpp>
+#include <range/v3/utility/optional.hpp>
+#include <range/v3/utility/semiregular.hpp>
+#include <range/v3/view/all.hpp>
+#include <range/v3/view/interface.hpp>
+#include <range/v3/view/reverse.hpp>
+#include <range/v3/view/view.hpp>
+
+namespace ranges
+{
+    /// \addtogroup group-views
+    /// @{
+    template<typename Rng, typename Pred>
+    struct trim_view : view_interface<trim_view<Rng, Pred>>
+    {
+    private:
+        Rng rng_;
+        semiregular_t<Pred> pred_;
+        detail::non_propagating_cache<iterator_t<Rng>> begin_;
+        detail::non_propagating_cache<iterator_t<Rng>> end_;
+
+    public:
+        CPP_assert(BidirectionalView<Rng> &&
+            IndirectUnaryPredicate<Pred, iterator_t<Rng>> &&
+            std::is_object<Pred>::value);
+
+        trim_view() = default;
+        trim_view(Rng rng, Pred pred)
+          : rng_(std::move(rng)), pred_(std::move(pred))
+        {}
+        CPP_template(typename R)(
+            requires BidirectionalRange<R> && ViewableRange<R> &&
+                Constructible<Rng, view::all_t<R>>)
+        trim_view(R &&rng, Pred pred)
+          : rng_(view::all(static_cast<R &&>(rng))), pred_(std::move(pred))
+        {}
```

Comment:
Nuke this constructor, please. Perhaps you saw similar constructors on the views in C++20. Those constructors are similarly a mistake and will be removed (I hope).